Formula/qt.rb

```diff
@@ -24,39 +24,35 @@ class Qt < Formula
     sha256 "25c4a693c787860b090685ac5cbeea18128d4d6361eed5b1bfed1b16ff6e4494" => :mojave
   end
 
-  keg_only "Qt 5 has CMake issues when linked"
+  keg_only "Qt has CMake issues when linked"
 
+  depends_on "cmake" => [:build, :test]
+  depends_on "ninja" => :build
   depends_on "pkg-config" => :build
+
   depends_on xcode: :build
-  depends_on macos: :sierra
+  depends_on macos: :mojave
 
   uses_from_macos "bison"
   uses_from_macos "flex"
+  uses_from_macos "icu4c"
   uses_from_macos "sqlite"
+  uses_from_macos "zlib"
 
-  # Find SDK for 11.x macOS
-  # Upstreamed, remove when Qt updates Chromium
+  # QTBUG-89176: Fix a critical bug in Qt 6.0.0 - QCache related crash.
+  # Patch from https://codereview.qt-project.org/c/qt/qtbase/+/326060
   patch do
-    url "https://raw.githubusercontent.com/Homebrew/formula-patches/92d4cf/qt/5.15.2.diff"
-    sha256 "fa99c7ffb8a510d140c02694a11e6c321930f43797dbf2fe8f2476680db4c2b2"
-  end
-
-  # Patch for qmake on ARM
-  # https://codereview.qt-project.org/c/qt/qtbase/+/327649
-  if Hardware::CPU.arm?
-    patch do
-      url "https://raw.githubusercontent.com/Homebrew/formula-patches/9dc732/qt/qt-split-arch.patch"
-      sha256 "36915fde68093af9a147d76f88a4e205b789eec38c0c6f422c21ae1e576d45c0"
-      directory "qtbase"
-    end
+    url "https://code.qt.io/cgit/qt/qtbase.git/patch/?id=b97001aa1cbd21008ebc48fe61b15fbcacb14875"
+    sha256 "87675faadfca6d2f77877328e69a464718efd9d289c1632ce536a5e3ebdc2157"
+    directory "qtbase"
   end
 
   def install
     args = %W[
-      -verbose
       -prefix #{prefix}
       -release
       -opensource -confirm-license
```

Comment:
The license parameters are ignored for Qt 6. So these can just be removed. See:

From https://doc.qt.io/qt-6/qt6-buildsystem.html:

> "We removed both selections in Qt 6. The choice of licenses that are available to you depends on any legal contracts you have with The Qt Company, not on any artifacts.
> 
> The configure options -opensource, -commercial, and -confirm-license are ignored in Qt 6, and might be removed in future versions of Qt."

Reply:
After running configurate script on my computer, I found some parameters have no effect..., such as `-pkg-config` which will translate to INPUT_pkg_config, the configure script output showed pkg-config is not enabled, but use the cmake counterpart `FEATURE_pkg_config=ON` is OK, I don't know this is my problem or the configurate script problem...😢